cli/command/image/trust.go

```diff
@@ -355,7 +350,7 @@ func TagTrusted(ctx context.Context, cli command.Cli, trustedRef reference.Canon
 	familiarRef := reference.FamiliarString(ref)
 	trustedFamiliarRef := reference.FamiliarString(trustedRef)
 
-	fmt.Fprintf(cli.Out(), "Tagging %s as %s\n", trustedFamiliarRef, familiarRef)
+	fmt.Fprintf(cli.Err(), "Tagging %s as %s\n", trustedFamiliarRef, familiarRef)
```

Comment:
Note sure whether this should go to stderr. This is the success message.

Reply:
This is the fix for the issue raised in #631.

The convention we should follow for stdout vs stderr error is:
* `stdout` is for "normal program output"
* `stderr` is for everything else (logging, warnings, etc)

This function is called from 3 commands. For both `image build` and `container create` this is definitely not normal program output, it's logging, so stderr is definitely correct.

For the last (`image pull`) it is a little less clear, but I think it could still easily be considered logging, since the desired action was "pull", not "tag".

Reply:
I guess that depends on whether one expects command "X" to be different to command "X with trust." In my mind, we should send some minimal amount of info to stdout such that if I pipe stderr to /dev/null I can still determine the operation behaved as desired.

Reply:
This is not how posix programs are supported to work. If we send **anything** to stdout we break anyone trying to use `docker build` and `docker run` in a script, when used with trust. The only output to stdout should be normal program output, which is the output from the command being run by `docker run`.

There is already a mechanism that allows you to determine if the operation behaved correctly or not, the exit status code. This PR also fixes some problems where the exit status was not being used correctly.